Arvelader.

fredforrest opened the pull request:
funktionalitet er der. Har dog tilføjet email, da det giver mening at arvelader(e) burde få tilsendt pdf på den mail de indtaster under udarbejdelse af testamentet? 

Ift de andre properties der kan findes på figma, så synes jeg ikke det gav mening at tilføje en masse informationer, når det hele blot handler om at få et overblik over sin arv til los familios. Make sense? 💯 🗡️ også jf. Max' snak om hvad der var vigtig.

Styling synes jeg sgu er god! (self 🦖 ) Farver og tema med grå divs går igen fra det i aktiver, så der er en rød tråd på det.

fredforrest commented:
This pull request introduces several new features and improvements to the `Arvefordeleren` project, focusing on the management of testators' information. The key changes include the addition of a new page for testators, corresponding styling, a repository for managing testator data, and updates to the solution file to support these changes.

### New Features:

* **New Testators Page:**
  * [`Components/Pages/Testators.razor`](diffhunk://#diff-9b4c8189f645fb044352221eb4c8e269da5563f89985cba2a2bc9bc947c02db5R1-R92): Added a new page for managing testators, including a form for adding new testators and a table for displaying existing ones.
  * [`Components/Pages/Testators.razor.css`](diffhunk://#diff-e4369038934204fe1ded9632ad76b024ded9a2cfbe054ea52e123e446292f8e2R1-R133): Added styling for the new testators page, including styles for the main container, headliner, body container, table, and form elements.

### Data Management:

* **Testator Repository:**
  * [`Models/Repositories/TestatorRepository.cs`](diffhunk://#diff-be8aa514f96b966d987f18b60187bde966e48c9db4cbc14a135b1bac33f45e03R1-R33): Introduced a static repository for managing testator data, including methods for adding, retrieving, and deleting testators.
  * [`Models/Testator.cs`](diffhunk://#diff-8851677422d8ae804ffbd2435abd82768db09ecdac30e79be512f6790e54917eR1-R18): Defined the `Testator` model with properties for `Id`, `Name`, `Address`, and `Email`.

### Project Configuration:

* **Solution File Update:**
  * [`Arvefordeleren.sln`](diffhunk://#diff-9ba915b85452ab80344b60ea9bc114b7e7105c08150f98b89c31993450b0ec56L3-R6): Updated the solution file to Visual Studio Version 17 and added a new section for solution properties. [[1]](diffhunk://#diff-9ba915b85452ab80344b60ea9bc114b7e7105c08150f98b89c31993450b0ec56L3-R6) [[2]](diffhunk://#diff-9ba915b85452ab80344b60ea9bc114b7e7105c08150f98b89c31993450b0ec56R19-R21)

NillasKA commented:
> 
> 
> This pull request introduces several new features and improvements to the `Arvefordeleren` project, focusing on the management of testators' information. The key changes include the addition of a new page for testators, corresponding styling, a repository for managing testator data, and updates to the solution file to support these changes.
> 
> 
> 
> ### New Features:
> 
> 
> 
> * **New Testators Page:**
> 
>   * [`Components/Pages/Testators.razor`](diffhunk://#diff-9b4c8189f645fb044352221eb4c8e269da5563f89985cba2a2bc9bc947c02db5R1-R92): Added a new page for managing testators, including a form for adding new testators and a table for displaying existing ones.
> 
>   * [`Components/Pages/Testators.razor.css`](diffhunk://#diff-e4369038934204fe1ded9632ad76b024ded9a2cfbe054ea52e123e446292f8e2R1-R133): Added styling for the new testators page, including styles for the main container, headliner, body container, table, and form elements.
> 
> 
> 
> ### Data Management:
> 
> 
> 
> * **Testator Repository:**
> 
>   * [`Models/Repositories/TestatorRepository.cs`](diffhunk://#diff-be8aa514f96b966d987f18b60187bde966e48c9db4cbc14a135b1bac33f45e03R1-R33): Introduced a static repository for managing testator data, including methods for adding, retrieving, and deleting testators.
> 
>   * [`Models/Testator.cs`](diffhunk://#diff-8851677422d8ae804ffbd2435abd82768db09ecdac30e79be512f6790e54917eR1-R18): Defined the `Testator` model with properties for `Id`, `Name`, `Address`, and `Email`.
> 
> 
> 
> ### Project Configuration:
> 
> 
> 
> * **Solution File Update:**
> 
>   * [`Arvefordeleren.sln`](diffhunk://#diff-9ba915b85452ab80344b60ea9bc114b7e7105c08150f98b89c31993450b0ec56L3-R6): Updated the solution file to Visual Studio Version 17 and added a new section for solution properties. [[1]](diffhunk://#diff-9ba915b85452ab80344b60ea9bc114b7e7105c08150f98b89c31993450b0ec56L3-R6) [[2]](diffhunk://#diff-9ba915b85452ab80344b60ea9bc114b7e7105c08150f98b89c31993450b0ec56R19-R21)

Er du ok? :D

fredforrest commented:
> > This pull request introduces several new features and improvements to the `Arvefordeleren` project, focusing on the management of testators' information. The key changes include the addition of a new page for testators, corresponding styling, a repository for managing testator data, and updates to the solution file to support these changes.
> > ### New Features:
> > 
> > * **New Testators Page:**
> >   
> >   * [`Components/Pages/Testators.razor`](https://github.com/NillasOrg/Arvefordeleren/pull/19/files#diff-9b4c8189f645fb044352221eb4c8e269da5563f89985cba2a2bc9bc947c02db5R1-R92): Added a new page for managing testators, including a form for adding new testators and a table for displaying existing ones.
> >   * [`Components/Pages/Testators.razor.css`](https://github.com/NillasOrg/Arvefordeleren/pull/19/files#diff-e4369038934204fe1ded9632ad76b024ded9a2cfbe054ea52e123e446292f8e2R1-R133): Added styling for the new testators page, including styles for the main container, headliner, body container, table, and form elements.
> > 
> > ### Data Management:
> > 
> > * **Testator Repository:**
> >   
> >   * [`Models/Repositories/TestatorRepository.cs`](https://github.com/NillasOrg/Arvefordeleren/pull/19/files#diff-be8aa514f96b966d987f18b60187bde966e48c9db4cbc14a135b1bac33f45e03R1-R33): Introduced a static repository for managing testator data, including methods for adding, retrieving, and deleting testators.
> >   * [`Models/Testator.cs`](https://github.com/NillasOrg/Arvefordeleren/pull/19/files#diff-8851677422d8ae804ffbd2435abd82768db09ecdac30e79be512f6790e54917eR1-R18): Defined the `Testator` model with properties for `Id`, `Name`, `Address`, and `Email`.
> > 
> > ### Project Configuration:
> > 
> > * **Solution File Update:**
> >   
> >   * [`Arvefordeleren.sln`](https://github.com/NillasOrg/Arvefordeleren/pull/19/files#diff-9ba915b85452ab80344b60ea9bc114b7e7105c08150f98b89c31993450b0ec56L3-R6): Updated the solution file to Visual Studio Version 17 and added a new section for solution properties. [[1]](https://github.com/NillasOrg/Arvefordeleren/pull/19/files#diff-9ba915b85452ab80344b60ea9bc114b7e7105c08150f98b89c31993450b0ec56L3-R6) [[2]](https://github.com/NillasOrg/Arvefordeleren/pull/19/files#diff-9ba915b85452ab80344b60ea9bc114b7e7105c08150f98b89c31993450b0ec56R19-R21)
> 
> Er du ok? :D

CoPilot resumé 😃

MiladFahham commented on Components/Pages/Testators.razor:
```diff
@@ -0,0 +1,92 @@
+
+@code {
+    private Testator testator = new Testator();
+    private List<Testator> testators = new List<Testator>();
+
+    private void AddTestator()
+    {
+        // Tilføj testator til listen og sessionbaseret lager (eller database, hvis det er tilfældet)
+
+        testators.Add(new Testator { Name = testator.Name, Address = testator.Address, Email = testator.Email });
+        testator = new Testator(); // Nulstil formularen efter indsendelse
+    }
```
Der vil nok forekomme problemer ved at anvende en lokal liste på din page frem for den statiske du oprettede i dit repository. 

F.eks. hvad sker der med instanserne af testator i din lokale liste når brugeren forlader pagen?

fredforrest replied:
der sker gode sager

MiladFahham commented on Models/Repositories/TestatorRepository.cs:
```diff
@@ -0,0 +1,33 @@
+﻿namespace Arvefordeleren.Models.Repositories
+{
+    public static class TestatorRepository
+    {
+        private static List<Testator> testators = new List<Testator>();
+        
+
+
+        public static void AddNewTestator(Testator testator)
+        {
+            var maxId = testators.Max(t => t.Id);
+            testator.Id = maxId + 1;
+            testators.Add(testator);
+        }
```
Du anvender ikke metoden AddNewTestator når ny Testator skal tilføjes i din razor page.

Du anvender ikke den oprettede liste der ligger i dit repository.

fredforrest replied:
true dat

MiladFahham reviewed (changes requested):
I forhold til design af brugergrænseflade, har jeg svært ved at se input-felterne når man skal indtaste oplysninger. Det kan dog blot være min computer og vi snakker bare om det i morgen.
